pkg/scanners/azure/arm/scanner.go

```diff
@@ -108,7 +113,7 @@ func (s *Scanner) initRegoScanner(srcFS fs.FS) error {
 	}
 	regoScanner := rego.NewScanner(types.SourceCloud, s.scannerOptions...)
 	regoScanner.SetParentDebugLogger(s.debug)
-	if err := regoScanner.LoadPolicies(s.loadEmbedded, srcFS, s.policyDirs, s.policyReaders); err != nil {
+	if err := regoScanner.LoadPolicies(s.loadEmbeddedLibraries, s.loadEmbeddedPolicies, srcFS, s.policyDirs, s.policyReaders); err != nil {
```

Comment:
nit: maybe pass `s` (Scanner) to LoadPolicies method as there are too many params

Reply:
Yeah that's a good idea, however `s` could be any type of Scanner. I tried to implement this with generics but quickly ran into a dead end that a method on a receiver where the base (in this case rego scanner) isn't generic cannot accept generic parameters. https://go.googlesource.com/proposal/+/refs/heads/master/design/43651-type-parameters.md#No-parameterized-methods

I will revisit this again. It will be a lot cleaner if we only passed `s` since all function inputs (except srcFS) come from `s`.